File: 1-100/28.implement-strstr.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        size_h = len(haystack)
        size_n = len(needle)
        if size_n == 0:
            return 0
        elif size_n > size_h:
            return -1
        
        for i in range(size_h-size_n+1):
            if haystack[i:i+size_n] == needle:
                return i
        
        return -1

File: 1-100/28.implement-strstr.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int size_h = haystack.size();
        int size_n = needle.size();
        if (size_n == 0)
        {
            return 0;
        }
        else if (size_n > size_h)
        {
            return -1;
        }

        for (int i = 0; i < size_h - size_n + 1; ++i)
        {
            bool bFlag = true;
            for (int j = 0; j < size_n; ++j)
            {
                if (haystack[i + j] != needle[j])
                {
                    bFlag = false;
                    break;
                }
            }

            if (bFlag)
            {
                return i;
            }
        }

        return -1;
    }
};